pkg/certs/certs.go

```diff
@@ -26,6 +30,65 @@ type Certificate struct {
 	Key          crypto.PrivateKey
 }
 
+// ParseCertificateBytes converts PEM formatted certificate to the Certificate struct.
+// Key is allowed to be empty, as there is cases when we're not storing it (e.g. CA bundles).
+func ParseCertificateBytes(certBytes, keyBytes []byte) (*Certificate, error) {
+	certs, err := cert.ParseCertsPEM(certBytes)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing certificates: %v", err)
+	}
+
+	var key crypto.PrivateKey
+	if len(keyBytes) != 0 {
+		keyPairCert, err := tls.X509KeyPair(certBytes, keyBytes)
+		if err != nil {
+			return nil, err
+		}
+		key = keyPairCert.PrivateKey
+	}
+
+	return &Certificate{certs, key}, nil
+}
+
+// FilterExpiredCerts checks are all certificates in the bundle valid, i.e. they have not expired.
+// The function returns new bundle with only valid certificates or error if no valid certificate is found.
+func FilterExpiredCerts(certs ...*x509.Certificate) ([]*x509.Certificate, error) {
+	okay := false
+	currentTime := time.Now()
+	var validCerts []*x509.Certificate
+	for _, c := range certs {
+		if c.NotAfter.After(currentTime) {
+			validCerts = append(validCerts, c)
+		}
+		okay = true
+	}
+
+	if !okay {
+		return nil, fmt.Errorf("no valid certificate found")
```

Comment:
This looks weird.  If you're filtering, why wouldn't you just return an empty slice?  That provides the same information to the caller and matches the method name.